_posts/2019-08-09-iglu-scala-client-0.6.0-released/index.md

```diff
@@ -0,0 +1,116 @@
+3. [New Validator](#new-validator)
+4. [Acknowledgements](#acknowledgements)
+5. [Roadmap and upcoming features](#roadmap)
+6. [Getting help](#help)
+
+<!--more-->
+
+<h3 id="api-changes">1. API changes</h3>
+
+Since 0.6.0 Iglu Scala Client exposes a new class called Client which consists of two independent entities: Resolver and Validator. Resolver is responsible for schema resolution, caching and error handling and Validator receives resolved schema with datum user wants to validate and returns the validation report. These entities can be used separately or even re-defined by user, but it is recommended to use Client class as an abstraction for most common use case - validation of self-describing entities.
+
+The Client class defines only one function:
+
+{% highlight scala %}
+def check[F[_]: RegistryLookup: Clock: Monad, A](instance: SelfDescribingData[A]): EitherT[F, ClientError, Unit]
+{% endhighlight %}
+
+Where:
+
+* F[_] is an abstract effect type, requiring a [tagless final][tagless-final] capabilities RegistryLookup and `cats.effect.Clock` as well as instance of cats.Monad type class. Two most common concrete effect types are cats.effect.IO and Id, all necessary machinery for them provided out-of-box, but user can also define this machinery for ZIO, Monix Task or sophisticated test type
+* A is a type of self-describing entity, such as JSON. Since 0.6.0, Iglu Client is based primarily on [circe][circe] Json type, but we're trying to leave it generic whenever possible
+* ClientError is a possible unsuccessful outcome, either at resolution or validation step. Unlike ProcessingMessage from pre-0.6.0 it provides type-safe and well-structured information about the failure. This type is widely used in upcoming Snowplow bad rows
+
+From this very short excerpt an astute Scala developer might notice that we replaced several libraries with their modern counterparts:  
+
+* [Circe][circe] is used instead of [Json4s][json4s]. Circe provides much better performance characteristics, does not rely on runtime reflection, provides very clean idiomatic API and remains one of the most popular JSON libraries in Scala ecosystem for last couple of years
+* [Cats Effect][cats-effect] is used for managing side-effects, instead of implicit effect management. One can use Iglu Client 0.6.0 without bothering about Cats Effect, but it is highly recommended in async-heavy environments
+* [Cats][cats] is used instead of [Scalaz][scalaz] as FP library of choice. Cats is a transitive dependency of Circe and Cats Effect, which makes it a natural choice, since no other dependencies are shipped with Scalaz.
+* [networknt json-schema-validator][networknt-validator] is used instead of FGE JSON Validator. One more change library, driving our bad rows effort.  networknt json-schema-validator is actively maintained, provides clean API and shows very impressive results in benchmarks
+
+You can find more usage examples on dedicated [wiki page][iglu-client-docs].
```

Comment:
You can find more usage examples on this dedicated [wiki page][iglu-client-docs].